src/components/Header.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
+import { ErrorMessage } from '../types/Error';
+
+type Props = {
+  onAddTodo: (title: string) => Promise<void>;
+  allCompletedTodos: boolean;
+  setError: React.Dispatch<React.SetStateAction<ErrorMessage | null>>;
+  fieldTitle: React.RefObject<HTMLInputElement>;
+  onChangeStatus: () => void;
+  hasTodos: boolean;
+};
+
+export const Header: React.FC<Props> = ({
+  onAddTodo,
+  allCompletedTodos,
+  setError,
+  fieldTitle,
+  onChangeStatus,
+  hasTodos,
+}) => {
+  const [newTitle, setNewTitle] = useState('');
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  useEffect(() => {
+    fieldTitle.current?.focus();
+  }, [isSubmitting, fieldTitle]);
+
+  const handleOnSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const title = newTitle.trim();
+
+    if (!title) {
+      setError(ErrorMessage.title);
+    } else {
+      setIsSubmitting(true);
+      onAddTodo(title)
+        .then(() => {
+          setNewTitle('');
+          setIsSubmitting(false);
+        })
+        .catch(() => {
+          setError(ErrorMessage.add);
+          setIsSubmitting(false);
```

Comment:
`setIsSubmitting(false` called in then and catch
will be better move to finally block of promise like this:
```
onAddTodo(title)
  .then(() => {
    setNewTitle('');
  })
  .catch(() => {
    setError(ErrorMessage.add);
  })
  .finally(() => {
    setIsSubmitting(false);
  });
```